### ISSUE TITLE
[DevTools] Support collapsing the entire component tree

### ISSUE THREAD

**Issue by samwinslow** [facebook/react#22074]
Similar to #16463, I find myself overwhelmed when trying to navigate the React component tree because everything is expanded by default.

- IMO there should be a simple way to recursively collapse everything and view the tree "top-down" rather than "bottom-up," just as VSCode's [-] button allows you to do.

<img width="116" alt="Screen Shot 2021-08-11 at 2 21 56 PM" src="https://user-images.githubusercontent.com/4645779/129082565-1f9b0a70-5f3f-434c-821d-51eebecc72d9.png">

- It would also be sensible to add vertical lines below each component's expand/collapse handle, because it becomes hard to see which elements are siblings in the tree when it's nested e.g. 15 layers deep.

<img width="507" alt="Screen Shot 2021-08-11 at 2 23 18 PM" src="https://user-images.githubusercontent.com/4645779/129082742-7eed8623-d23d-44b2-902f-e4efef30ed4f.png">

**Comment by bvaughn** [comment]
> IMO there should be a simple way to recursively collapse everything and view the tree "top-down" rather than "bottom-up," just as VSCode's [-] button allows you to do.

There is. Just click the arrow to the left of the root node, or use the left arrow key.

https://user-images.githubusercontent.com/29597/129084368-289241cf-7576-49d3-85ae-b77cc9b30ac4.mp4

> It would also be sensible to add vertical lines below each component's expand/collapse handle, because it becomes hard to see which elements are siblings in the tree when it's nested e.g. 15 layers deep.

I tried this during the design exploration phase, but didn't end up leaving it in because it looked bad when indentation was small:
https://twitter.com/brian_d_vaughn/status/1134862518420090886

Instead, we highlight the subtree:
https://twitter.com/brian_d_vaughn/status/1134834828459687937

**Comment by bvaughn** [comment]
I think we just have different expectations.

The Component tab is more similar to the browser's Elements tab, which functions the same way as DevTools does. (It does not recursively collapse).

**Comment by samwinslow** [comment]
> The Component tab is more similar to the browser's Elements tab, which functions the same way as DevTools does. (It does not recursively collapse).

Right, it doesn't recursively collapse by default, but that functionality is available in parallel to the typical collapse behavior via a separate UI element. It's a bit harder to find, but it's super useful once you discover it.

<img width="347" alt="Screen Shot 2021-08-11 at 4 05 46 PM" src="https://user-images.githubusercontent.com/4645779/129098501-1566006e-1bfb-41e9-be96-d10c4928ae23.png">

Thank you for the quick responses, by the way!
